### PR TITLE
cleanup(w1r3): return `error` in go functions

### LINKED ISSUE #42: Prefer `return err` over `log.Fatalf()` in w1r3/go benchmark
I see a lot of use of `log.Fatal` outside of main -- I think this is fine for a prototype but ultimately it's more typical in Go to have functions return errors and then fail from main if need be. This is helpful for testing (assuming we add tests at some point) and for being able to understand the chain of calls.

In this case you at least should definitely include the underlying error in the format string for `log.Fatalf` (elsewhere it looks fine though).

_Originally posted by @tritone in https://github.com/googleapis/storage-shared-benchmarking/pull/40#discussion_r1573028135_

### PR DESCRIPTION
**Pull request by coryan**
Prefer returning an `error` over directly calling
`log.Fatalf()`.

Fixes #42

### CODE REVIEW

**Comment by coryan** [comment]
@tritone can you take a look?

File: w1r3/go/main.go [hunk 1]
```diff
@@ -119,20 +119,38 @@ func main() {
...
+		log.Fatalf("Error enabling Cloud Trace exporter %v", err)
+	}
 	defer cleanupMeter()
 
-	transports := makeTransports(ctx, transportArgs)
+	transports, err := makeTransports(ctx, transportArgs)
+	if err != nil {
+		log.Fatalf("Error creating transports %v", err)
+	}
+	for _, t := range transports {
```
Comment (tritone) [inline w1r3/go/main.go:137]: This check can be skipped; there should never be a nil error and a nil client.

**Review by tritone — APPROVED** [review]
One suggestion, otherwise looks good

File: w1r3/go/main.go (hunk 1 above)
Comment (coryan) [reply w1r3/go/main.go]: Done.

**Pull request merged**